app/client/ui/ActiveUserList.ts

```diff
@@ -0,0 +1,162 @@
+    userImages.reverse();
+
+    return cssActiveUserList(
+      ...userImages.map(image => dom('li', image)),
+      { "aria-label": t("active user list") },
+    );
+  });
+}
+
+function createUserIndicator(user: Partial<FullUser>, options = { overlapLeft: false }) {
+  return createUserListImage(
+    user,
+    hoverTooltip(user.name, { key: "topBarBtnTooltip" }),
+    options.overlapLeft ? createStyledUserImage.cls("-overlapping") : undefined,
+    { 'aria-label': `${t('active user')}: ${user.name}`}
+  );
+}
+
+function createRemainingUsersIndicator(users: Partial<FullUser>[], userCount?: number) {
+  const count = userCount ?? users.length;
+  return cssRemainingUsersButton(
+    cssRemainingUsersImage(
+      `+${count}`,
+      cssUserImage.cls("-medium"),
+      dom.style("font-size", "12px"),
+    ),
+    menu(
+      () => users.map(user => remainingUsersMenuItem(
+        createUserImage(user, 'medium'),
+        user.name,
+      )),
+      {
+        // Avoids an issue where the menu code will infinitely loop trying to find the
+        // next selectable option, when using keyboard navigation, due to having none.
+        allowNothingSelected: true,
+      }
+    ),
+    { 'aria-label': t('open full active user list') },
+  );
+}
+
+// Flex-direction is reversed to give us the correct overlaps without messing with z-indexes.
+const cssActiveUserList = styled('ul', `
+  display: flex;
+  align-items: center;
+  justify-content: end;
+  list-style: none;
+
+  margin: 0;
+  padding: 0;
+  border: 0 solid;
+
+  flex-direction: row-reverse;
+`);
+
+const userImageBorderCss = `
+  border: 2px solid ${components.topHeaderBg};
+  box-sizing: content-box;
+`;
+
+const createStyledUserImage = styled(createUserImage, `
+  ${userImageBorderCss};
+
+  &-overlapping {
+    margin-left: -4px;
+  }
+`);
+
+const createUserListImage = (user: Parameters<typeof createUserImage>[0], ...args: DomElementArg[]) =>
+  createStyledUserImage(
+    user,
+    'medium',
+    cssUserImage.cls('-reduced'),
+    dom.hide(isXSmallScreenObs()),
+    ...args
+  );
+
+const cssRemainingUsersImage = styled(cssUserImage, `
+  margin-left: -4px;
+  background-color: ${components.userListRemainingUsersBg};
+  ${userImageBorderCss};
+`);
+
+const cssRemainingUsersButton = styled('button', `
+  margin: 0;
+  padding: 0;
+  border: 0 solid;
+  font: inherit;
+  letter-spacing: inherit;
+  color: inherit;
+  border-radius: 0;
+  background-color: transparent;
+  opacity: 1;
+  appearance: button;
+`);
+
+export const remainingUsersMenuItem = styled(`div`, `
+  display: flex;
+  justify-content: flex-start;
+  padding: var(--weaseljs-menu-item-padding, 8px 24px);
+  align-items: center;
+  color: ${theme.menuItemFg};
+  --icon-color: ${theme.accentIcon};
+  text-transform: none;
+
+  & > :first-child {
+    margin-right: 5px;
+  }
+`);
+
+function compareUserProfiles(a: VisibleUserProfile, b: VisibleUserProfile) {
+  if (!a.isAnonymous && b.isAnonymous) {
```

Comment:
(Unimportant comment) You could do `nativeCompare(a.isAnonymous, b.isAnonymous) || nativeCompare(a.name, b.name)` -- `nativeCompare` is defined in `app/common/gutil.ts` (this is following my own advice in https://mill.plainopen.com/why-sorting-is-harder-than-it-seems)